source/nodejs/adaptivecards/src/card-elements.ts

```diff
@@ -4351,6 +4351,148 @@ export class ColumnSet extends CardElementContainer {
     }
 }
 
+export class Source {
+    mimeType: string;
+    url: string;
+
+    parse(json: any): Source {
+        this.mimeType = json["mimeType"];
+        this.url = json["url"];
+        return this;
+    }
+
+    toJSON() {
+        return { mimeType: this.mimeType, url: this.url };
+    }
+}
+
+export class Media extends CardElement {
+    poster: string;
+    altText: string;
+    sources: Array<Source> = [];
+    private _mediaType: string;
+    private _mediaElement: HTMLMediaElement;
+    static readonly acceptedMediaTypes = ["audio", "video"];
+
+    private processSources(): Array<Source> {
```

Comment:
This might be my personal taste, but I find this a bit more complex and hard to read than probably necessary (especially the type filter that modifies itself). How about something like this?

```
let goodSources: Array<MediaSource> = [];

for (let source of this.sources) {
	let mimeComponents = source.mimeType.split('/');

	if (mimeComponents.length == 2) {
		if (!this._mediaType) {
			let index = Media.acceptedMediaTypes.indexof(mimeComponents[0]);

			if (index >= 0) {
				this._mediaType = Media.acceptedMediaTypes[index];
			}
		}

		if (mimeComponent[0] == this._mediaType) {
			goodSourdces.push(source);
		}
	} 
}

return goodSources;
```

Reply:
Sure can change it.